apigentools/commands/generate.py

```diff
@@ -354,6 +354,8 @@ def pull_repository(self, language, branch=None):
             )
             if branch is not None:
                 try:
+                    run_command(["git", "fetch", "origin", branch], cwd=output_dir)
+                    run_command(["git", "branch", branch, "FETCH_HEAD"], cwd=output_dir)
```

Comment:
Actually, won't this create a new branch instead of tracking an existing one ?

Reply:
no it will use the fetched head